Mathlib/Algebra/Homology/DerivedCategory/Ext/Map.lean

```diff
@@ -0,0 +1,130 @@
+
+open Limits Abelian
+
+variable {C : Type u} [Category.{v} C] [Abelian C]
+variable {D : Type u'} [Category.{v'} D] [Abelian D]
+
+variable (F : C ⥤ D) [F.Additive] [PreservesFiniteLimits F] [PreservesFiniteColimits F]
+
+open Localization
+
+instance [h : HasExt.{w'} D] (X Y : C) : HasSmallLocalizedShiftedHom.{w'}
+    (HomologicalComplex.quasiIso D (ComplexShape.up ℤ)) ℤ
+    (((F ⋙ CochainComplex.singleFunctor D 0).obj X))
+    (((F ⋙ CochainComplex.singleFunctor D 0).obj Y)) :=
+  h (F.obj X) (F.obj Y)
+
+/-- The map between `Ext` induced by `F.mapShiftedHomAddHom`. -/
+noncomputable def Abelian.Ext.mapExactFunctor [HasExt.{w} C] [HasExt.{w'} D] {X Y : C} {n : ℕ}
+    (f : Ext.{w} X Y n) : Ext.{w'} (F.obj X) (F.obj Y) n :=
+  (F.mapHomologicalComplexUpToQuasiIsoLocalizerMorphism
+    (ComplexShape.up ℤ)).smallShiftedHomMap
+    ((F.mapCochainComplexSingleFunctor 0).app X) ((F.mapCochainComplexSingleFunctor 0).app Y) f
+
+open Functor in
+lemma Abelian.Ext.mapExactFunctor_eq_shiftedHom_map
+    [HasDerivedCategory.{t} C] [HasDerivedCategory.{t'} D]
+    [HasExt.{w} C] [HasExt.{w'} D] {X Y : C} {n : ℕ} (e : Ext X Y n) :
+    (e.mapExactFunctor F).hom =
+    (F.mapDerivedCategorySingleFunctor 0).inv.app X ≫ e.hom.map F.mapDerivedCategory ≫
+    ((F.mapDerivedCategorySingleFunctor 0).hom.app Y)⟦(n : ℤ)⟧' := by
+  rw [← ShiftedHom.comp_mk₀ _ 0 rfl, ← ShiftedHom.mk₀_comp 0 rfl]
+  simp only [Ext.hom, Ext.homEquiv, comp_obj]
+  apply Eq.trans ((F.mapHomologicalComplexUpToQuasiIsoLocalizerMorphism
+    (ComplexShape.up ℤ)).equiv_smallShiftedHomMap
+    DerivedCategory.Q DerivedCategory.Q
+    ((F.mapCochainComplexSingleFunctor 0).app X) ((F.mapCochainComplexSingleFunctor 0).app Y)
+    F.mapDerivedCategory F.mapDerivedCategoryFactors.symm e)
+  simp only [comp_obj, mapHomologicalComplexUpToQuasiIsoLocalizerMorphism_functor, Iso.app_inv,
+    Iso.symm_hom, Iso.symm_inv, Iso.app_hom]
+  congr 2
+  · simp only [mapCochainComplexSingleFunctor, mapDerivedCategorySingleFunctor,
+      DerivedCategory.singleFunctorIsoCompQ, isoWhiskerRight_refl, isoWhiskerLeft_refl,
+      Iso.refl_trans, Iso.trans_inv, Iso.refl_inv, Category.id_comp, isoWhiskerRight_inv,
+      Iso.symm_inv, Category.assoc, isoWhiskerLeft_inv, NatTrans.comp_app, comp_obj, Iso.refl_symm,
+      associator_inv_app, whiskerRight_app, associator_hom_app, whiskerLeft_app, Category.comp_id]
+    exact (Category.id_comp _).symm
+  · congr 1
+    simp only [mapCochainComplexSingleFunctor, mapDerivedCategorySingleFunctor,
+      DerivedCategory.singleFunctorIsoCompQ, isoWhiskerRight_refl, isoWhiskerLeft_refl,
+      Iso.refl_symm, Iso.refl_trans, Iso.trans_hom, isoWhiskerLeft_hom, Iso.symm_hom,
+      isoWhiskerRight_hom, Iso.refl_hom, NatTrans.comp_app, comp_obj, associator_hom_app,
+      whiskerLeft_app, associator_inv_app, whiskerRight_app, NatTrans.id_app, Category.id_comp]
+    nth_rw 2 [← Category.assoc]
+    exact (Category.comp_id _).symm.trans (Category.id_comp _).symm
+
```

Comment:
As the variables `[HasExt.{w} C] [HasExt.{w'} D] (X Y : C) (n : ℕ)` are used until the end, you could start a section here with these variables.